src/main.rs

```diff
@@ -38,11 +45,8 @@ fn lookup(x: char) -> usize {
     }
 }
 
-// get the three-letter equivalent of an amino acid
-// aa (char): the amino acid to translate, eg 'A' -> "Ala"
-// returns: String
-fn three_letter_code(aa: char) -> String {
-    let three_letter_map: HashMap<char, &str> = [
+static THREE_LETTER_CODE: Lazy<HashMap<char, &'static str>> = Lazy::new(|| {
```

Comment:
This is a big change compared to how you did it, but it allows to build a single copy (in memory) of the `THREE_LETTER_CODE` HashMap that can be reused without building it again (like it was being done in the `three_letter_code` function). This is not a large optimization for a small program, but makes a lot of difference if this needs to be executed many times. Because building a `HashMap` is not possible in `const` contexts (to be able to build it during compile time) yet, I added a dependency on the [`once_cell`](https://crates.io/crates/once_cell) crate, which allows the initialization of this variable in this context during runtime (but only once, when it is first accessed).

Reply:
Ah, I was wondering how to create a static map!